resources/gnomad_liftover.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+from gnomad_hail.resources import *
+from gnomad_hail.utils import *
+import logging
+
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("gnomAD_liftover")
+logger.setLevel(logging.INFO)
+
+
+def get_checkpoint_path(data_type: str) -> str:
+    """
+    Creates a checkpoint path for Table
+    :param str data_type: Data type (exomes or genomes)
+    :return: Output checkpoint path
+    :rtype: str
+    """
+
+    return f'gs://seqr-datasets/methods_dev/test_data/kc/gnomAD_{data_type}_sites.ht'
+
+
+def lift_ht(ht: hl.Table, data_type: str, rg: hl.genetics.ReferenceGenome) -> hl.Table:
+    """
+    Lifts gnomAD release table from b37 to b38
+    :param Table ht: gnomAD release Table
+    :param str data_type: Data type (exome or genome)
+    :param ReferenceGenome rg: Reference genome, b38
+    :return: gnomAD release Table lifted to b38
+    :rtype: Table
+    """
+
+    # annotate release file with liftover coordinates
+    ht = ht.annotate(new_locus = hl.liftover(ht.locus, rg, include_strand=True),
+                     old_locus=ht.locus
+                    )
+    ht = ht.key_by(locus = ht.new_locus.result, alleles = ht.alleles)
+    ht = ht.checkpoint(get_checkpoint_path(data_type), overwrite=True)
+    return ht
+
+
+def snp_mismatch(ht: hl.Table, data_type: str, rg: hl.genetics.ReferenceGenome) -> hl.Table:
+    """
+    Annotates mismatches between reference allele and allele in reference fasta
+    :param Table ht: Table of SNPs to be annotated
+    :param str data_type: Data type (exome or genome)
+    :param ReferenceGenome rg: GRCh38 reference genome with fasta sequence loaded
+    :return: Table annotated with mismatches between reference allele and allele in fasta
+    :rtype: Table
+    """
+ 
+    # filter to snps
+    ht = ht.filter(hl.is_snp(ht.alleles[0], ht.alleles[1]))
+    snp_count = ht.count()
+    logger.info('{} SNPs in table'.format(snp_count))
+
+    # check if reference allele matches what is in reference fasta
+    # for snps on negative strand, make sure reverse complement of ref allele matches what is in reference fasta
+    ht = ht.annotate(
+        reference_mismatch = hl.cond(
+                                ht.new_locus.is_negative_strand,
+                                (generic.flip_base(ht.alleles[0]) != hl.get_sequence(ht.locus.contig, ht.locus.position, reference_genome=rg)),
```

Comment:
here locus is already the lifted over locus;  new_locus is a struct: 
```'new_locus': struct {
        result: locus<GRCh38>,
        is_negative_strand: bool
    }```
would it be clearer if I used new_locus.result.contig instead of locus?